src/pages/resultsView/download/DownloadUtils.ts

```diff
@@ -321,6 +322,36 @@ export function generateOtherMolecularProfileDownloadData(
         : [];
 }
 
+export function generateGenericAssayProfileData(
+    molecularProfileId: string[],
+    unfilteredCaseAggregatedData?: CaseAggregatedData<GenericAssayData>
+): { [key: string]: GenericAssayData[] } {
+    const sampleFilter = (data: GenericAssayData) => {
+        return molecularProfileId.includes(data.molecularProfileId);
+    };
+
+    return unfilteredCaseAggregatedData
```

Comment:
I think we can combine `generateGenericAssayProfileData` and `generateGenericAssayDataByStableId`